src/utils/privileges.cpp

```diff
@@ -26,6 +28,21 @@
 namespace my_cap {
 
 #ifdef ANTINET_linux
+static void change_user_if_root() {
+	uid_t uid = getuid();
+	if (uid != 0) return; // not root
+	const char* sudo_user_env = std::getenv("SUDO_USER");
+	if (sudo_user_env == nullptr) throw std::runtime_error("Cannot read SUDO_USER env") ; // get env error
+	std::string sudo_user_name(sudo_user_env);
+	struct passwd *pw = getpwnam(sudo_user_name.c_str());
+	if (pw == nullptr) throw std::runtime_error("getpwnam error");
+	uid_t normal_user_uid = pw->pw_uid;
+	assert(normal_user_uid != 0);
```

Comment:
this can happen when root does sudo ./program (I tested in bash)
In this case,  we should throw custom exception no_regular_user_to_exit_sudo("The sudo was called by root");
this is common thing to happen in scripts